enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -0,0 +1,1156 @@
+         >
+         > This should be caught by *`prev-boot-commit.system` is `unhealthy` > `current-commit` != `prev-boot-commit` >
+         > `current-commit` does not exist in `health.json`*
+         * Exit 1, block `microshift run`
+
+   - `current-commit == prev-boot-commit`
+     > Because `current-commit` is already present in `health.json`, it's 2nd, 3rd,... boot of the commit (not 1st).
+
+     - `backups/current-commit.id/` exists
+       > commit was healthy at least once already.
+       > It means greenboot removed `boot_counter` and current boot isn't due to greenboot's automatic reboot.
+       > After addressing issues, admin should mark commit as healthy, so this shouldn't happen.
+       > Let's assume admin addressed the issues but forgot to mark as healthy and just rebooted the device.
+       * No backup, no restore, skip migration, allow `microshift run`
+         > If admin addressed issues, it should be healthy again.
+         > If issue persists, it'll end up unhealthy requiring manual intervention
+         > (which is the same result if we'd block `microshift run`).
+
+     - `backups/current-commit.id/` doesn't exist
+       > commit was always unhealthy - it wasn't "accepted" by greenboot, so it might still rollback.
+       > Because admin should heal the commit before manually rebooting, we assume it's reboot due to greenboot.
+
+       - `earlier-commit not in health.json`
+         > For MicroShift it's first commit it's running, but it doesn't mean it's the only commit
+         > (rollback commit might be without MicroShift).
+         >
+         > ostree status:
+         > 1. `current-commit` == `prev-boot-commit` - currently running,
+         >    last boot unhealthy, greenboot didn't accept it yet so it might still rollback
+         > 1. (optionally) rollback (possibly without MicroShift)
+         * Delete data, try from clean state
+
+       - `earlier-commit in health.json`
+         > For MicroShift it looks like it ran already on the device, but `earlier-commit` might or might not be
+         > the rollback commit.
+         >
+         > ostree status:
+         > 1. `current-commit` == `prev-boot-commit` - currently running, last boot unhealthy, greenboot didn't accept it yet
+         > 1. (auto) rollback (might be `earlier` or not)
+
+         - `earlier-commit == ostree rollback commit`
+           > ostree status:
+           > 1. `current` == `prev-boot` - currently running, last boot unhealthy, greenboot didn't accept it yet
+           > 1. `earlier` - (auto) rollback
+           >
+           > Let's sum it up: it's 2nd or 3rd boot of `current-commit` and it's `unhealthy` since 1st boot.
+           > It was staged directly from `earlier` which also ran MicroShift.
+
+           - `earlier-commit` was healthy
+             - `microshift/version.json[commit] == earlier-commit.id`
```

Comment:
What does this notation mean? The commit recorded in the last version file is the same as the `earlier-commit`?

Reply:
Yes, field `commit` from `version.json` matches id of `earlier-commit`. But this will change during enh. simplification